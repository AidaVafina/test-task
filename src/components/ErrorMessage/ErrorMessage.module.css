.container {
    background-color: var(--color-error-bg, #ffe6e6);
    color: var(--color-error-text, #cc0000);
    padding: 1rem;
    border-radius: 8px;
    margin: 1rem 0;
    text-align: center;
}

.text {
    margin: 0 0 0.5rem;
    font-weight: bold;
}

.button {
    background-color: var(--color-error-button-bg, #cc0000);
    color: #fff;
    border: none;
    padding: 0.5rem 1rem;
    cursor: pointer;
    border-radius: 4px;
    font-size: 0.9rem;
}

.button:hover {
    background-color: #a30000;
}
