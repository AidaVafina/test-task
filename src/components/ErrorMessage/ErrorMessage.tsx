import React from 'react';
import styles from './ErrorMessage.module.css';

interface ErrorMessageProps {
    message: string;
    onRetry?: () => void;
}

const ErrorMessage: React.FC<ErrorMessageProps> = ({ message, onRetry }) => {
    return (
        <div className={styles.container}>
            <p className={styles.text}>Ошибка: {message}</p>
            {onRetry && (
                <button className={styles.button} onClick={onRetry}>
                    Повторить
                </button>
            )}
        </div>
    );
};

export default ErrorMessage;
