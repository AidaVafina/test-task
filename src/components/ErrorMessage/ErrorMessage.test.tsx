import { render, fireEvent } from '@testing-library/react';
import { describe, it, beforeEach, expect, vi } from 'vitest';
import ErrorMessage from './ErrorMessage';

describe('ErrorMessage Component', () => {
    const mockMessage = 'Test error message';
    const mockOnRetry = vi.fn();

    beforeEach(() => {
        mockOnRetry.mockClear();
    });

    it('should match snapshot with retry button', () => {
        const { container } = render(
            <ErrorMessage message={mockMessage} onRetry={mockOnRetry} />
        );
        expect(container).toMatchSnapshot();
    });

    it('should match snapshot without retry button', () => {
        const { container } = render(
            <ErrorMessage message={mockMessage} />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render error message', () => {
        const { container } = render(
            <ErrorMessage message={mockMessage} onRetry={mockOnRetry} />
        );
        expect(container.textContent).toContain(mockMessage);
    });

    it('should call onRetry when button is clicked', () => {
        const { container } = render(
            <ErrorMessage message={mockMessage} onRetry={mockOnRetry} />
        );
        const button = container.querySelector('button');
        
        if (button) {
            fireEvent.click(button);
        }

        expect(mockOnRetry).toHaveBeenCalledTimes(1);
    });
}); 