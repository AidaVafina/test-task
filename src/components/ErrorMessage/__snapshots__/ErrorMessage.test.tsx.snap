// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage Component > should match snapshot with retry button 1`] = `
<div>
  <div
    class="_container_0c75d4"
  >
    <p
      class="_text_0c75d4"
    >
      Ошибка: 
      Test error message
    </p>
    <button
      class="_button_0c75d4"
    >
      Повторить
    </button>
  </div>
</div>
`;

exports[`ErrorMessage Component > should match snapshot without retry button 1`] = `
<div>
  <div
    class="_container_0c75d4"
  >
    <p
      class="_text_0c75d4"
    >
      Ошибка: 
      Test error message
    </p>
  </div>
</div>
`;

exports[`ErrorMessage Component should match snapshot with retry button 1`] = `
<div>
  <div
    class="container"
  >
    <p
      class="text"
    >
      Ошибка: 
      Test error message
    </p>
    <button
      class="button"
    >
      Повторить
    </button>
  </div>
</div>
`;

exports[`ErrorMessage Component should match snapshot without retry button 1`] = `
<div>
  <div
    class="container"
  >
    <p
      class="text"
    >
      Ошибка: 
      Test error message
    </p>
  </div>
</div>
`;
